src/libexpr/eval.cc

```diff
@@ -2326,10 +2333,31 @@ StorePath EvalState::copyPathToStore(NixStringContext & context, const SourcePat
 
 SourcePath EvalState::coerceToPath(const PosIdx pos, Value & v, NixStringContext & context, std::string_view errorCtx)
 {
-    auto path = coerceToString(pos, v, context, errorCtx, false, false, true).toOwned();
-    if (path == "" || path[0] != '/')
-        error("string '%1%' doesn't represent an absolute path", path).withTrace(pos, errorCtx).debugThrow<EvalError>();
-    return CanonPath(path);
+    try {
```

Comment:
This changes this method to exclude `__toString`.
Is this a (breaking) change compared to Nix versions before `SourcePath`?

Reply:
Indeed it didn't handle __toString. I've fixed it and added a test.